Classes_Homework_5/fromxml.py

```diff
@@ -0,0 +1,141 @@
+import xml.etree.ElementTree as ET
+
+from counts import Counts
+from modules.combine import Combine
+from modules.dates import Dates
+from modules.file import Files
+from modules.input import DateInput
+from exec_utils.configloader import Config
+from fromjson import FromJson
+from modules.exceptions import NoSectionsError, PastDate, NoValue
+from modules.run import Execute
+
+# instantiate classes
+cnf = Config()
+cnt = Counts(cnf.get_values("PATHS", "target_file"), cnf.get_values("PATHS", "csv_words"), cnf.get_values("PATHS", "csv_letters"))
+f = Files()
+e = Execute()
+dte = Dates()
+di = DateInput(cnf.get_values("ERRORS", "past_date"),
+               cnf.get_values("PATTERNS", "date_format"),
+               dte.get_current_date())
+
+# declare variables
+label1, label2, label3 = cnf.get_values("LABELS", "news_label"), \
+                         cnf.get_values("LABELS", "ad_label"), \
+                         cnf.get_values("LABELS", "recipe_label")
+decor, decor_length = cnf.get_values("RESTRICTIONS", "n"), \
+                      cnf.get_values("RESTRICTIONS", "count_n")
+label, city, txt, date, cal = cnf.get_values("PROPERTIES", "label"), \
+                              cnf.get_values("PROPERTIES", "city"), \
+                              cnf.get_values("PROPERTIES", "text"), \
+                              cnf.get_values("PROPERTIES", "date"), \
+                              cnf.get_values("PROPERTIES", "kcal")
+
+
+class FromXML(FromJson):  # inherit methods from FromJson in fromjson.py.
+    def get_xml(self, file_path):
+        """Parse xml by path"""
+        doc = ET.parse(file_path)
+        return doc
+
+    def get_sections(self, file_path):
+        """Split xml by keyword into sections, in our case "sectionLable"""
+        doc = self.get_xml(file_path)
+        sections = doc.findall('./'+label)  # find all elements with sectionLabel tag
+        if sections:
+            return sections
+        else:
+            raise NoSectionsError(label)
```

Comment:
The "label" isn't in the list of function arguments, so the function can fail when it uses outside the file.